File: BASIC_OF_ALGORITHM/How_to_Solve_N_&_M/NM_4.cpp
//비내림차순에서의 수열 (중복가능)
#include <iostream>
using namespace std;

int A[10];

void go(int index, int start, int n, int m) {
	if (index == m) {
		for (int i = 0; i < m; i++) {
			cout << A[i] << " ";
		}
		cout << endl;
		return;
	}
	for (int i = start; i <= n; i++) {
		A[index] = i;
		go(index + 1, i, n, m);
	}

}


int main() {
	int N, M;
	cin >> N >> M;

	go(0, 1, N, M);










	return 0;
}

File: BASIC_OF_ALGORITHM/How_to_Solve_N_&_M/NM_4.py
A = [0] * 10
def go(index, start, n, m):
    if index == m:
        for i in range(m):
            print(A[i], end = " ")
        print()
        return
    
    for i in range(start, n + 1):
        A[index] = i
        go(index + 1, i, n, m)



n,m = input().split()

go(0, 1, int(n), int(m))

